test/e2e/apimachinery/custom_resource_definition.go

```diff
@@ -42,29 +51,153 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
 
 			randomDefinition := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
 
-			//create CRD and waits for the resource to be recognized and available.
+			// Create CRD and waits for the resource to be recognized and available.
 			randomDefinition, err = fixtures.CreateNewCustomResourceDefinition(randomDefinition, apiExtensionClient, f.DynamicClient)
-			if err != nil {
-				e2elog.Failf("failed to create CustomResourceDefinition: %v", err)
-			}
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
 
 			defer func() {
 				err = fixtures.DeleteCustomResourceDefinition(randomDefinition, apiExtensionClient)
-				if err != nil {
-					e2elog.Failf("failed to delete CustomResourceDefinition: %v", err)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
+			}()
+		})
+
+		/*
+			Release : v1.16
+			Testname: Custom Resource Definition, list
+			Description: Create a API extension client, define 10 random custom resource definitions and list them using a label selector. API server MUST be able to list the custom resource definitions and delete them via delete collection.
+		*/
+		ginkgo.It("listing custom resource definition objects works ", func() {
+			testListSize := 10
+			config, err := framework.LoadConfig()
+			framework.ExpectNoError(err, "loading config")
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing apiExtensionClient")
+
+			// Label the CRDs we create so we can list only them even though they are cluster scoped
+			testUUID := string(uuid.NewUUID())
+
+			// Create CRD and wait for the resource to be recognized and available.
+			crds := make([]*v1beta1.CustomResourceDefinition, testListSize)
+			for i := 0; i < testListSize; i++ {
+				crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+				crd.Labels = map[string]string{"e2e-list-test-uuid": testUUID}
+				crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+				framework.ExpectNoError(err, "creating CustomResourceDefinition")
+				crds[i] = crd
+			}
+
+			// Create a crd w/o the label to ensure the label selector matching works correctly
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
+			}()
+
+			selectorListOpts := metav1.ListOptions{LabelSelector: "e2e-list-test-uuid=" + testUUID}
+			list, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().List(selectorListOpts)
+			framework.ExpectNoError(err, "listing CustomResourceDefinitions")
+			gomega.Expect(len(list.Items)).To(gomega.Equal(testListSize))
```

Comment:
We can use `framework.ExpectEqual(len(list.Items), testListSize)` instead of `gomega.Expect(len(list.Items)).To(gomega.Equal(testListSize))`.
Then we can remove importing `gomega` in this test module.
In addition, after merging https://github.com/kubernetes/kubernetes/pull/80785 this code will be blocked with the additional code check.

Reply:
Removed. I still have a `gomega.Expect(expected).To(gomega.Not(gomega.BeNil()))` below. Is that correct or is there a `framework` function I should favor?

Reply:
Nevermind, just saw your next comment.